Tests/RoutingKitTests/RouterPerformanceTests.swift

```diff
@@ -0,0 +1,147 @@
+            router.register(route: Route(path: [
+                .constant(letter),
+                .parameter("\(letter)_id")
+            ], output: letter))
+        }
+        
+        measure {
+            var params = Parameters()
+            for _ in 0..<100_000 {
+                _ = router.route(path: ["aaaaaaag", "42"], parameters: &params)
+            }
+        }
+    }
+    
+    public func testMinimalRouterCaseSensitivePerformance() throws {
+        guard performance(expected: 0.021) else { return }
+        let router = TrieRouter.init(String.self, options: [.caseInsensitive])
+        for letter in ["a"] {
+            router.register(route: Route(path: [
+                .constant(letter)
+            ], output: letter))
+        }
+        
+        measure {
+            var params = Parameters()
+            for _ in 0..<100_000 {
+                _ = router.route(path: ["a"], parameters: &params)
+            }
+        }
+    }
+    
+    public func testMinimalRouterCaseInsensitivePerformance() throws {
+        guard performance(expected: 0.016) else { return }
+        let router = TrieRouter.init(String.self)
+        for letter in ["a"] {
+            router.register(route: Route(path: [
+                .constant(letter)
+            ], output: letter))
+        }
+        
+        measure {
+            var params = Parameters()
+            for _ in 0..<100_000 {
+                _ = router.route(path: ["a"], parameters: &params)
+            }
+        }
+    }
+    
+    
+    public func testMinimalEarlyFailPerformance() throws {
+        guard performance(expected: 0.013) else { return }
+        let router = TrieRouter.init(String.self)
+        for letter in ["aaaaaaaaaaaaaa"] {
+            router.register(route: Route(path: [
+                .constant(letter)
+            ], output: letter))
+        }
+        
+        measure {
+            var params = Parameters()
+            for _ in 0..<100_000 {
+                _ = router.route(path: ["baaaaaaaaaaaaa"], parameters: &params)
+            }
+        }
+    }
+
+    public static let allTests = [
+        ("testCaseSensitivePerformance", testCaseSensitivePerformance),
+        ("testCaseInsensitivePerformance", testCaseInsensitivePerformance),
+        ("testCaseInsensitiveRoutingMatchFirstPerformance", testCaseInsensitiveRoutingMatchFirstPerformance),
+        ("testCaseInsensitiveRoutingMatchLastPerformance", testCaseInsensitiveRoutingMatchLastPerformance),
+        ("testMinimalRouterCaseInsensitivePerformance", testMinimalRouterCaseInsensitivePerformance),
+        ("testMinimalRouterCaseSensitivePerformance", testMinimalRouterCaseSensitivePerformance),
+        ("testMinimalEarlyFailPerformance", testMinimalEarlyFailPerformance),
+    ]
+}
+
+func performance(expected seconds: Double, name: String = #function) -> Bool {
+    guard !_isDebugAssertConfiguration() else {
```

Comment:
Planning on removing this now that we're filtering tests within circle config unless there's another reason to keep it around.

Reply:
@twof i like having that there for testing on macOS. The performance tests take a lot of time and it's kidn of annoying when I'm used to hitting cmd+U regularly during development.

Reply:
Another thing you could do is uncheck the perf tests under the package scheme, but I'm guess that gets wiped out when you regenerate the xcodeproj. 

<img width="665" alt="screen shot 2019-02-21 at 4 11 10 pm" src="https://user-images.githubusercontent.com/5561501/53210627-5f144600-35f3-11e9-9cd9-886fc13c27f5.png">

Reply:
Yeah, those will get reset on xcodeproj (re)generation, also they don't have effect on Linux